server/src/main/java/org/elasticsearch/cluster/routing/allocation/DiskThresholdMonitor.java

```diff
@@ -350,9 +351,9 @@ protected void updateIndicesReadOnly(Set<String> indicesToUpdate, ActionListener
         Settings readOnlySettings = readOnly ? Settings.builder()
             .put(IndexMetadata.SETTING_READ_ONLY_ALLOW_DELETE, Boolean.TRUE.toString()).build() :
             Settings.builder().putNull(IndexMetadata.SETTING_READ_ONLY_ALLOW_DELETE).build();
-        client.admin().indices().prepareUpdateSettings(indicesToUpdate.toArray(Strings.EMPTY_ARRAY))
-            .setSettings(readOnlySettings)
-            .execute(ActionListener.map(wrappedListener, r -> null));
+        client.admin().indices().updateSettings(new UpdateSettingsRequest(readOnlySettings, indicesToUpdate.toArray(Strings.EMPTY_ARRAY)))
+            .thenApply(response -> (Void) null)
```

Comment:
Just curiosity, why is this needed?

Reply:
Because the ActionListener is using `<Void>`. It doesn't type check without it